flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/util/HBaseSerde.java

```diff
@@ -225,6 +217,13 @@ public Get createGet(Object rowKey) {
 
     /** Converts HBase {@link Result} into {@link RowData}. */
     public RowData convertToRow(Result result) {
+        // The output rows needs to be initialized each time
```

Comment:
Could we add a `reuse` flag for this function so that we can improve the performance when user didn't enable the cache or used in the HBaseRowDataInputFormat

Reply:
@leonardBang  Sounds good. I think if we add a reuse flag for this function, will complicate the method.
Maybe we could add a method specifically for the cache. Or you have a better idea?

Reply:
Both above ideas make sense to me,  just choice your favor